qiita_pet/handlers/websocket_handlers.py

```diff
@@ -70,3 +72,26 @@ def on_close(self):
         yield Task(self.toredis.unsubscribe, self.channel)
         self.r_client.delete('%s:messages' % self.channel)
         self.redis.disconnect()
+
+
+class SelectSamplesHandler(WebSocketHandler, BaseHandler):
+    @authenticated
+    def on_message(self, msg):
+        # When the websocket receives a message from the javascript client,
+        # parse into JSON
+        msginfo = loads(msg)
+        num_samples = sum(len(x) for x in msginfo["samples"])
+        default = Analysis(self.current_user.default_analysis)
+        if msginfo["action"] == "select":
+            # match proc data with samples and add them
+            select = {}
+            for pid, samples in zip(msginfo["proc_data"], msginfo["samples"]):
+                select[pid] = samples
+                default.add_samples(select)
+            self.write_message("%d samples selected" % num_samples)
+        elif msginfo["action"] == "deselect":
+            for pid, samples in zip(msginfo["proc_data"], msginfo["samples"]):
+                default.remove_samples([pid], samples)
```

Comment:
Is there a reason you mofied `add_samples` to take a dict of list, but left remove_samples as it was before? It would be nice if these two functions took the same parameter type